cmd/kubeadm/app/cmd/join.go

```diff
@@ -607,7 +607,7 @@ func fetchInitConfigurationFromJoinConfiguration(cfg *kubeadmapi.JoinConfigurati
 	// Perform the Discovery, which turns a Bootstrap Token and optionally (and preferably) a CA cert hash into a KubeConfig
 	// file that may be used for the TLS Bootstrapping process the kubelet performs using the Certificates API.
 	glog.V(1).Infoln("[join] Discovering cluster-info")
-	tlsBootstrapCfg, err := discovery.For(cfg)
+	tlsBootstrapCfg, err := discovery.For(cfg, kubeadmconstants.DefaultClusterName)
```

Comment:
what if the already existing control-plane node is not using the default cluster name?

Reply:
This is the first step to fetch the `ClusterConfiguration`, not sure I understand the question. Right after this and after fetching the `ClusterConfiguration` with this temporary kubeconfig the final one will be set here: https://github.com/kubernetes/kubernetes/pull/70757/files#diff-0960dc0bb7c3e7113a0daa027856b8f9R623